Week2/homework/exercise5.js

```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+let drinkTray = [ ];
+const drinkTypes = ['cola','lemonade','water']
+console.log(drinkTypes) /*just tested if the array is working properly*/
+
+for (let i = 0; drinkTray.length <= 5; i++) {
+    var idx = drinkTypes[Math.floor(Math.random() * drinkTypes.length)]; /*wanted to pick a random piece of the array*/
+    drinkTray.push(drinkTypes[idx]);
+    console.log(drinkTray);
+    break
+}
+
+console.log("Hey guys, I brought" + drinkTray + "!")
```

Comment:
A few things here:
1. The logic of generating random drink types and adding them to the drink tray doesn't seem to be achieving the desired result. Reason why I say this is because it could be that the random drink generated is always cola or always lemonade. The requirement, however, was that we shouldn't be able to add more than 2 drinks of the same type.
2. Also, the break statement at the end on line 11 doesn't make much sense because the for loop is going to move to the next iteration anyways irrespective of the `break` keyword.
3. The reason why you see a bunch of commas is because `drinkTray` is an array and that's how the elements inside the array are separated.
4. Also, would recommend that you be consistent with using the keywords `var`, `let` or `const` and try to avoid `var` as much as possible. 🙂

Reply:
I am seriously struggling with this one and I fixed part of it but not the whole thing yet. That's why I didn't update this file for now, I'll keep working until I can fully understand the logic behind it.

Reply:
Try to keep it simple to begin with. Instead of fetching random drinks from the list, try to put all the list items incrementally in the tray. Since we have only 3 drinks and 5 spots in the tray, if you add them in a straight sequence, you will see that you can have only 2 drinks of same kind atmost.

Once this is achieved, you can try to add more logic to it. But start with as simple a solution as possible. 🙂

Reply:
Yash, I am out of ideas to solve this one once for all. I got at the point where I there is no more than two types of drink at the same time into the array but there is this another rule, that asks that water just shows one time that I can't acchieve.